battleShip.py

```diff
@@ -0,0 +1,280 @@
+        pickle.dump(obj1, file)
+        pickle.dump(obj2, file)
+    print("\t\tGame saved successfully :>)")
+
+def loadGame(): # Load the game
+    try:
+        list=[]
+        with open("savegame.pickle", "rb") as file:
+            p1=pickle.load(file)
+            list.append(p1)
+            p2=pickle.load(file)
+            list.append(p2)
+            return list
+    except FileNotFoundError:
+        print("No saved game found :<(")
+        return None
+def main():
+    while True:
+        displayMenu()
+        choice = input("\t\tEnter your choice: ")
+
+        if choice == '1':
+            while True:
+                displayMenu2()
+                choice = input("\t\tEnter your choice: ")
+                if choice == '1':
+                    p = input("\t\tPlayer 1 Enter your name :) ")
+                    p2 = input("\t\tPlayer 2 Enter your name :) ")
+                    player1 = Player(p)
+                    player2 = Player(p2)
+
+                    player1.createGrid(26)
+                    player2.createGrid(26)
+
+                    player1.placeShips(6, 7, 26)
+                    player2.placeShips(6, 7, 26)
+
+                    while True:
+                        print(f"\n\t\t---- {player1.name} Turn ----")
+                        print(f"\t\t {player1.name} Grid: \n")
+                        player1.displayGrid()
+                        player1.attack(player2)
+
+                        if not player2.ships:
+                            print(f"\t\t\t{player1.name} wins! :>)")
+                            break
+
+                        print(f"\n\t\t---- {player2.name} Turn ----")
+                        print(f"\t\t{player2.name} Grid: \n")
+                        player2.displayGrid()
+                        player2.attack(player1)
+
+                        if not player1.ships:
+                            print(f"\t\t\tKudos {player2.name} You wins! :>)")
+                            break
+                        while True:
+                            print("\n\t\t Do you want to save the game? (y/n) :>")
+                            print("\t\t Quit the game? (q) :>s")
+                            saveChoice = input("\n\t\t Enter your choice:> ")
+                            if saveChoice.lower() == 'y':
+                                saveGame(player1, player2)
+                            elif saveChoice.lower() == 'n':
+                                break
+                            elif saveChoice.lower() == 'q':
+                                exit()
+                            else:
+                                print("\t\tInvalid choice. Try again :<")
+                                continue
+                elif choice == '2':
+                    p = input("\t\tEnter your name :) ")
+                    player1 = Player(p)
+                    player2 = Player("Computer")  # Create a computer player
+
+                    player1.createGrid(26)
```

Comment:
both players need to have ships on same board, and current player should be able see only his own ships and tiles that he has destroyed.